src/main/resources/test_files/runtime/FunctionArray/FunctionArrayDescendant5.iq

```diff
@@ -1,4 +1,4 @@
 (:JIQS: ShouldRun; Output="( )" :)
```

Comment:
What is the meaning of Output="( )"? Is it specific to the tests? A query returning an empty sequence should simply serialize it as an empty string on the command line.

Reply:
Normally, EmptySequenceItem should not make it to the outer result of the query. The resulting empty string is the consequence of the iterator returning hasNext()=false. No item is ever returned here.

Reply:
Multiple results which are empty sequences: this can never happen to any query :-) The result is always exactly one sequence. In such a case, for example the query "(), (), ()", the result is simply the empty sequence.

Reply:
Your first explanation makes sense to me, however there remains the problem that I've encountered earlier regarding FLOWR clauses sent to spark. In the closure example that I'm sending in the picture, the result.iterator() is passed to the framework. When all items contained by the iterator are null (which is the empty sequence case) the framework throws a NullPointerException. This was the reason why I came up with the EmptySequenceItem which acts as a placeholder for Java null, hence preventing the exception. Can you see another solution for this?
![image](https://user-images.githubusercontent.com/13052327/44528189-30bbb980-a6e9-11e8-9876-2c52a495c09c.png)

Yes that's definitely correct for the jsoniq. I'd assumed the test suite behaved differently and hence results of individual queries would be aggregated together in a single output of the form ( , , , ...) . Yet it applies for the test suite as well :) Thanks for pointing it out.

Reply:
If I understand you correctly, this will also get solved "for free" when hasNext() is fixed to return false. next() will thus never be called (and will thus never return a Java null either). Then, result will simply be the empty list, which is consistent with the empty sequence.

Does it make sense?

Reply:
Yes that might be the case for some queries but I'm afraid I doesn't fit all cases. For example the following image shows the descendant-array function:
![image](https://user-images.githubusercontent.com/13052327/44539229-e7c52e80-a703-11e8-99bf-7c45e206be23.png)
Iterator of the function itself is opened at least once, hence triggering the calculations. The contained sequence(sequenceIterator) may be an empty sequence by itself or may result in an empty sequence following the calculations of getDescendantArrays. However, in both cases the function iterator has to return an item to the closure, which we return as null. This null causes the NullPointerException. If the list itself is empty however, which I have implemented in the v2 branch(other PR), then it works as you've described above. In order to get the list empty, I have removed the null element from the closure.